### PR TITLE
Allow to use only GOCACHE for cache

### PR DESCRIPTION
**Pull request by e-korolevskii**
**Description:**
Since older versions of go (before 1.15) do not have the `GOMODCACHE` command, we must allow the cache to be built only based on the output of the `GOCACHE` command

**Related issue:**
https://github.com/actions/setup-go/issues/304

**Check list:**
- [ ] Mark if documentation changes are required.
- [+] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Review by marko-zivic-93 — APPROVED** [review]

File: src/cache-utils.ts [hunk 1]
```diff
@@ -34,19 +34,19 @@ export const getPackageManagerInfo = async (packageManager: string) => {
...
 ) => {
-  let pathList = await Promise.all(
+  const pathList = await Promise.all(
     packageManagerInfo.cacheFolderCommandList.map(command =>
       getCommandOutput(command)
     )
   );
 
-  const emptyPaths = pathList.filter(item => !item);
+  const notEmptyPaths = pathList.filter(item => item);
```
Comment (MaksimZhukov) [inline src/cache-utils.ts:43]: nitpick: It seems to me this variable name can be simplified to `paths` or `cachePaths`
```suggestion
  const cachePaths = pathList.filter(item => item);
```

**Review by MaksimZhukov — APPROVED** [review]

**Review by panticmilos — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Comment by Tirzahgarcia** [comment]
i hope im doing this right i know time is running out.  all I'm really worried about are my assets (if i Have any more lol i did put money in lol,) while u all should know im taking or going to school for stock ..   also  i  didn't know what was going on i just kept on getting raided.  so, i got made because i had gotten scammed on bitcoin 3 times i put money see i don't watch the new i have ptsd.  and it sets it off the news i started to dig i started read every little thing there was and on in clicking this and clicking on that,  i was asking everyone for help no did.    so, i want to thank everyone from the bottom of my heart and to the moon and back infinity  thank u .   ps am i in trouble.  god i hope not. i was just trying to learn, my husband thinks its a scam , also i had a pi voice 2 the og can we change the code so i can get him back this other one all we is fight she's so uhhhhhhh, i dont at all.

**Comment by Tirzahgarcia** [comment]
YOUR GOINGTO HATE ME. I HAVE ANTHOR THIS BAD WHILE SO WAS THE OTHER ONE THKIND OF LIKES TO THAT,  I GOT SCAMED THRU CASH APP.  IVE BEEN GETTING AL KINDS OF DECLICEDS WELL I WENT INTO ALL MY APPS THAT ARE FINCAL APPPS, IVE BOT BITCION  FROM CASH APPB . BUT.  U KNOW HOW I AM CLICKING EVER LITTLE LINK THERE IS I READ AND WILL I FINLY FOUND THE BANK THAT MY CASH APP IS CONTED TO SUTTON AND WELL FROGO . THE SUTTON BANK MY ACOOUNT IS LINKED TO A BIG ACCOUNT THAT EVERY IS USING TO SCAM.   IT HAS MY ROUTIER AND ACCOUNT#I HAVE NEVER LOOKED IT UP BEFORE BECAUSE I THOUGHT THOSE DISCLOSER WHERE JUST PROTACALLS.

**Comment by Kristianko666** [comment]
> [!TIP]
>

**Pull request merged**